File: payflow-admin-client/src/api/merchantNotify.ts
import request from './request'
import { DEFAULT_PAGE_SIZE } from '@/constants/pagination'

export interface MerchantNotifyListItem {
  notifyId: string
  orderId: string
  merchantId: string
  merchantOrderNo?: string
  notifyType: string
  notifyUrl?: string
  summaryStatus: string
  attemptCount: number
  lastAttemptAt?: string
  lastFailReason?: string
  lastResponsePreview?: string
  orderStatus?: string
  notifyPayloadStatus?: string
}

export interface MerchantNotifyListResponse {
  total: number
  page: number
  size: number
  list: MerchantNotifyListItem[]
}

export interface MerchantNotifyAttemptItem {
  attemptNo: number
  resultStatus: string
  failReasonType?: string
  failReasonDetail?: string
  httpStatus?: number
  durationMs?: number
  requestParams: Record<string, unknown> | string
  responseBody?: string
  truncated?: boolean
  createdAt?: string
}

export interface MerchantNotifyDetailResponse {
  summary: MerchantNotifyListItem & { createdAt?: string; updatedAt?: string }
  attempts: MerchantNotifyAttemptItem[]
}

export interface MerchantNotifyByOrderResponse {
  orderId: string
  orderStatus?: string
  summaries: MerchantNotifyListItem[]
}

export interface MerchantNotifyListQuery {
  merchantId?: string
  orderId?: string
  merchantOrderNo?: string
  notifyType?: string
  summaryStatus?: string
  startTime?: string
  endTime?: string
  page?: number
  size?: number
}

export const listMerchantNotifies = async (
  params: MerchantNotifyListQuery,
): Promise<MerchantNotifyListResponse> => {
  const data = (await request.get<MerchantNotifyListResponse>('/admin/merchant-notifies', {
    params,
  })) as unknown as MerchantNotifyListResponse
  return {
    total: Number(data?.total ?? 0),
    page: Number(data?.page ?? params.page ?? 1),
    size: Number(data?.size ?? params.size ?? DEFAULT_PAGE_SIZE),
    list: data?.list ?? [],
  }
}

export const getMerchantNotifyDetail = async (
  notifyId: string,
): Promise<MerchantNotifyDetailResponse> => {
  const data = (await request.get<MerchantNotifyDetailResponse>(
    `/admin/merchant-notifies/${encodeURIComponent(notifyId)}`,
  )) as unknown as MerchantNotifyDetailResponse
  return data
}

export const getMerchantNotifyByOrder = async (
  orderId: string,
  notifyType?: string,
): Promise<MerchantNotifyByOrderResponse> => {
  const data = (await request.get<MerchantNotifyByOrderResponse>(
    `/admin/merchant-notifies/by-order/${encodeURIComponent(orderId)}`,
    { params: notifyType ? { notifyType } : {} },
  )) as unknown as MerchantNotifyByOrderResponse
  return data
}

File: payflow-admin-server/src/main/java/com/payflow/admin/entity/cashier/MerchantNotify.java
package com.payflow.admin.entity.cashier;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("cashier_merchant_notify")
public class MerchantNotify {

    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("notify_id")
    private String notifyId;

    @TableField("order_id")
    private String orderId;

    @TableField("merchant_id")
    private String merchantId;

    @TableField("merchant_order_no")
    private String merchantOrderNo;

    @TableField("notify_type")
    private String notifyType;

    @TableField("notify_url")
    private String notifyUrl;

    @TableField("summary_status")
    private String summaryStatus;

    @TableField("attempt_count")
    private Integer attemptCount;

    @TableField("last_attempt_at")
    private LocalDateTime lastAttemptAt;

    @TableField("last_fail_reason")
    private String lastFailReason;

    @TableField("last_response_preview")
    private String lastResponsePreview;

    @TableField("order_status_snapshot")
    private String orderStatusSnapshot;

    @TableField("notify_payload_status")
    private String notifyPayloadStatus;

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;
}
